### DISCUSSION TITLE
Why does asPreview not exists in all types? How can I get previews for CPTs?

### DISCUSSION THREAD

**Discussion by JorgeSivil** [wp-graphql/wp-graphql#3186] [General] [answered]
Hello! I'm a newbie WP developer.

I have a CPT and I would like to implement Preview using NextJS.

My queries are custom, I'm not using Faust (I think, although I see it installed).

```
export const neighborhoodQuery = /* GraphQL */ `
    query neighborhoodQuery($uri: String!) {
        cptNeighborhoodBy(uri: $uri, asPreview: true) {
            isPreview
            ...CptNeighborhood
        }
    }
    ${seoFragment}
    ${breadcrumbFragment}
    ${neighborhoodFragment}
`;
```

But I'm getting an error  `GraphQLDocumentError: Unknown argument "asPreview" on field "RootQuery.cptNeighborhoodBy".`

I'm wondering what other query could I use to get the desired result?

**Comment by justlevine** [comment]
`{ObjectType}By` has been deprecated for several years. It should work if you use `cptNeighborhood( id: $uri, idType: URI, asPreview: true ) { ...MyFrag }`

**Reply by JorgeSivil** [reply]
@justlevine thank you, I have tried what you mentioned with this query

```
query Neighborhood {
  cptNeighborhood(id: "/neighborhoods/sunterra", idType: URI, asPreview: false) {
    isPreview
    neighborhoodEditor {
      fieldAcfgroupNeighborhoodsContact {
        cptNeighborhoodNewstarAddress1
      }
    }
  }
}
```

However, when I set asPreview to true, I get null as result.

```
{
  "data": {
    "cptNeighborhood": null
  },
  "extensions": {
    "debug": [],
    "queryAnalyzer": {
      "keys": "667275dcd392becacc7e6c241f241aca4632daa82858f1ae17c9ff53b5b88e7f graphql:Query operation:Neighborhood cG9zdDoyNjYx",
      "keysLength": 114,
      "keysCount": 4,
      "skippedKeys": "",
      "skippedKeysSize": 0,
      "skippedKeysCount": 0,
      "skippedTypes": []
    }
  }
}
```

This is the web IDE, so it should be authenticated isn't it?

What I did was

1) Go to the post
2) Change the address
3) Click Preview
4) Run the query

**Reply by justlevine** [reply, marked as answer]
@JorgeSivil only if you [toggle yourself as authenticated](https://www.wpgraphql.com/docs/wp-graphiql#toggle-between-public-and-authenticated-requests).
